groups/join_group_channel.js

```diff
@@ -0,0 +1,116 @@
+const asyncAuto = require('async/auto');
+const asyncMap = require('async/map');
+const {getMethods} = require('ln-service');
+const {getNodeAlias} = require('ln-sync');
+const {returnResult} = require('asyncjs-util');
+
+const joinChannelGroup = require('./join_channel_group');
+const getJoinDetails = require('./get_join_details');
+
+const formatNodes = arr => arr.join(', ');
+const isCode = n => !!n && n.length === 98;
+const niceName = ({alias, id}) => `${alias} ${id}`.trim();
+const signPsbtEndpoint = '/walletrpc.WalletKit/SignPsbt';
+
+/** Join a channel group
+
+  {
+    ask: <Ask Function>
+    code: <Group Invite Code String>
+    lnd: <Authenticated LND API Object>
+    logger: <Winston Logger Object>
+    max_rate: <Max Opening Fee Rate Number>
+  }
+
+  @returns via cbk or Promise
+*/
+module.exports = (args, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {    
+        if (!isCode(args.code)) {
+          return cbk([400, 'ExpectedValidJoinCodeToJoinGroup']);
+        }  
+
+        if (!args.lnd) {
+          return cbk([400, 'ExpectedAuthenticatedLndToJoinGroup']);
+        }
+
+        if (!args.logger) {
+          return cbk([400, 'ExpectedWinstonLoggerToJoinGroupp']);
+        }
+
+        if (!args.max_rate) {
+          return cbk([400, 'ExpectedMaxOpeningFeeRateToJoinGroup']);
+        }
+
+        return cbk();
+      },
+
+      // Get methods to confim partial signing is supported
+      getMethods: ['validate', ({}, cbk) => getMethods({lnd: args.lnd}, cbk)],
+
+      // Make sure that partially signing a PSBT is valid
+      confirmSigner: ['getMethods', ({getMethods}, cbk) => {
+        if (!getMethods.methods.find(n => n.endpoint === signPsbtEndpoint)) {
+          return cbk([400, 'ExpectedLndSupportingPartialPsbtSigning']);
+        }
+
+        return cbk();
+      }],
+
+      // Decode the code and get group details
+      getJoinDetails: ['confirmSigner', ({}, cbk) => {
+        return getJoinDetails({
+          code: args.code, 
+          lnd: args.lnd, 
+          logger: args.logger,
+          rate: args.max_rate
```

Comment:
not necessary?

Reply:
I just don't think this variable is ever referenced in that function, and that makes sense, like why would get group details as a function need to know about what the max auto-accepting fee rate would be